hub/util/exceptions.py

```diff
@@ -274,6 +279,11 @@ def __init__(self) -> None:
         )
 
 
+class HashlistDoesNotExistError(Exception):
+    def __init__(self):
+        super().__init__(f"Hashlist doesn't exist.")
```

Comment:
Use better exception messages that tell users what went wrong. In this case, probably tell them that you can't compare the datasets because either the first one or the second one wasn't created with hashing enabled.